### PR TITLE
Add chrony_drift.

### PR DESCRIPTION
**Pull request by thermes**
This commit adds a plugin for `chronyd driftfile` .

### CODE REVIEW

File: plugins/chrony/chrony_drift [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+#%# family=auto
+#%# capabilities=autoconf
+
+=cut
+
+driftfile=${driftfile:-'/var/lib/chrony/chrony.drift'}
+
+do_autoconf () {
+    if [ -r $driftfile ]; then
```
Comment (sumpfralle) [inline plugins/chrony/chrony_drift]: Please quote the variable here.

`shellcheck` is your friend :)

File: plugins/chrony/chrony_drift [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+graph_category time
+drift.label System Clock Gain/Loss Rate
+error_bound.label Estimate of Error Bound
+graph_info The rate at which the system clock gains or loses time relative to real time.
+EOM
+}
+
+
+do_ () {
+    if [ -r $driftfile ]; then
```
Comment (sumpfralle) [inline plugins/chrony/chrony_drift]: Same as above (quoting).
The same applies to the lines below.

File: plugins/chrony/chrony_drift [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        echo "drift.value $(cat $driftfile | awk '{print $1;}')"
```
Comment (sumpfralle) [inline plugins/chrony/chrony_drift]: Feel free to replace `cat FOO | awk ...` with `awk ... FOO`.

File: plugins/chrony/chrony_drift [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        echo "error_bound.value $(cat $driftfile | awk '{print $2;}')"
+    else
+        echo "Could not read driftfile '$driftfile'." >&2
```
Comment (sumpfralle) [inline plugins/chrony/chrony_drift]: In this case you should not error out, but respond with the value `U` (i.e. "unknown").

File: plugins/chrony/chrony_drift [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        exit 1
+    fi
+}
+
+
+case $1 in
+    autoconf|config|'')
+        do_$1
```
Comment (sumpfralle) [inline plugins/chrony/chrony_drift]: Please quote this one, too. (just `"do_$1"`)

File: plugins/chrony/chrony_drift [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        exit $?
+        ;;
+    *)
+        echo "Don't know how to do that" >&2
+        exit 1
+        ;;
+esac
+
```
Comment (sumpfralle) [inline plugins/chrony/chrony_drift]: This is safely a dead code branch. Please just remove the following lines.
(I understand your point, but the code above looks neat and clean - thus it is hard to run into surprises here)

**Comment by sumpfralle** [comment]
@thermes: ping?

**Comment by thermes** [comment]
@sumpfralle 
Thank you for the review.
The review points have been fixed. I also used `shellcheck`. Please check again.
(Sorry for the late response.)

File: plugins/chrony/chrony_drift [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+do_ () {
+    if [ -r "$driftfile" ]; then
+        echo "drift.value $(awk '{print $1;}' "$driftfile")"
+        echo "error_bound.value $(awk '{print $2;}' "$driftfile")"
+    else
+        echo "drift.value U"
+        echo "error_bound.value U"
+        exit 1
```
Comment (sumpfralle) [inline plugins/chrony/chrony_drift]: See #982 (same line).

File: plugins/chrony/chrony_drift [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    fi
+}
+
+
+case $1 in
+    autoconf|config|'')
+        do_"$1"
+        exit $?
```
Comment (sumpfralle) [inline plugins/chrony/chrony_drift]: See #982 (same line).

**Comment by thermes** [comment]
@sumpfralle
Thank you for the review.
Fixed the issue and squashed.

**Comment by sumpfralle** [comment]
Thank you!

**Pull request merged**